### PR TITLE
fixed issues from initial commit

### PR DESCRIPTION
**Pull request by Kanapriya**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>1.4.0</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${pom.artifactId}</Bundle-SymbolicName>
+                        <Export-Package>
```
Comment (Kanapriya) [inline pom.xml]: Fixed in https://github.com/wso2-extensions/esb-inbound-iso8583/pull/9

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                            org.wso2.carbon.inbound.iso8583.listening.*,
+                            org.jpos.iso.*,
+                            org.apache.commons.cli.*,
+                            org.jdom.*,
+                        </Export-Package>
+                        <Import-Package>
```
Comment (Kanapriya) [inline pom.xml]: Fixed in https://github.com/wso2-extensions/esb-inbound-iso8583/pull/9

File: src/main/java/org/wso2/carbon/inbound/iso8583/listening/ConnectionRequestHandler.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+import org.jpos.iso.ISOMsg;
+import org.jpos.iso.ISOPackager;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.Socket;
+
+/**
+ * Class for handling the iso message request.
```
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/inbound/iso8583/listening/ConnectionRequestHandler.java]: Fixed in https://github.com/wso2-extensions/esb-inbound-iso8583/pull/9

File: src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageInject.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+     * @param object iso fields.
+     */
+    public boolean inject(ISOMsg object) {
+        org.apache.synapse.MessageContext msgCtx = createMessageContext();
+        msgCtx.setProperty("inbound.endpoint.name", params.getName());
+        InboundEndpoint inboundEndpoint = msgCtx.getConfiguration().getInboundEndpoint(params.getName());
+        CustomLogSetter.getInstance().setLogAppender(inboundEndpoint.getArtifactContainerName());
+        msgCtx.setProperty(ISO8583Constant.ISO8583_INBOUND_MSG_ID, msgCtx.getMessageID());
+
+        if (injectingSeq == null || injectingSeq.equals("")) {
```
Comment (hmrajas) [inline src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageInject.java]: Use StringUtils.isEmpty() to check the string null or empty.

Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageInject.java]: Fixed in f49cbaa5cff19f162a1f6d3ddfb8a2e6a963e43d

File: src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageConnection.java [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+     * create the threadPool to handle the concurrent request.
+     */
+    private ExecutorService getExecutorService() {
+        Properties properties = params.getProperties();
+        String coreThreads = properties.getProperty(ISO8583Constant.INBOUND_CORE_THREADS);
+        String maxThreads = properties.getProperty(ISO8583Constant.INBOUND_MAX_THREADS);
+        String threadSafeTime = properties.getProperty(ISO8583Constant.INBOUND_THREAD_ALIVE);
+        String queueLength = properties.getProperty(ISO8583Constant.INBOUND_THREAD_QLEN);
+        try {
+            if ((!StringUtils.isEmpty(coreThreads)) && (!StringUtils.isEmpty(maxThreads)) &&
```
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageConnection.java]: Fixed in f49cbaa5cff19f162a1f6d3ddfb8a2e6a963e43d

File: src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageConnection.java [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+     *
+     * @param connection Socket connection
+     * @param params     the InboundProcessorParams
+     */
+    private void handleClientRequest(Socket connection, InboundProcessorParams params) {
+        try {
+            threadPool.execute(new ConnectionRequestHandler(connection, params));
+        } catch (RejectedExecutionException re) {
+            // If the pool is full complete the execution with the same thread
+            log.warn("Worker pool has reached the maximum capacity.");
```
Comment (pamod) [inline src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageConnection.java]: is there any reason for having a warn log here, shouldn't we increase the severity to error ? also could you print the stack trace as well

File: src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583ReplySender.java [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+            if (resourceStream != null)
+                prop.load(resourceStream);
+        } catch (IOException e) {
+            handleException("Could not find the properties file ", e);
+        } finally {
+            try {
+                if (resourceStream != null)
+                    resourceStream.close();
+            } catch (IOException e) {
+                log.error("Couldn't close the inputStream");
```
Comment (pamod) [inline src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583ReplySender.java]: shall we print the stack trace ?

File: src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageConnection.java (hunk 6 above)
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageConnection.java]: Just to warn the user that queue is full and  cannot add any more threads in the thread pool.
I have added the stack trace as well, please find that in following commit
7bdcc0493fa7623ea576d32be2c8b292f098467e

File: src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583ReplySender.java (hunk 7 above)
Comment (Kanapriya) [inline src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583ReplySender.java]: Fixed in 7bdcc0493fa7623ea576d32be2c8b292f098467e

File: src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageInject.java [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+                .getSequence(injectingSeq);
+        try {
+            OMElement parentElement = messageBuilder(object);
+            msgCtx.getEnvelope().getBody().addChild(parentElement);
+            ISO8583ReplySender replySender = new ISO8583ReplySender(connection);
+            replySender.sendBack(msgCtx);
+            if (seq != null) {
+                seq.setErrorHandler(onErrorSeq);
+                if (log.isDebugEnabled()) {
+                    log.info("injecting message to sequence : " + injectingSeq);
```
Comment (jagathsisira) [inline src/main/java/org/wso2/carbon/inbound/iso8583/listening/ISO8583MessageInject.java]: Should be a debug log instead of info.

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+                            !javax.xml.namespace,
+                            javax.xml.namespace;
+                            version=0.0.0,
+                            *;resolution:=optional,
```
Comment (Niranjan-K) [inline pom.xml]: As a best practice, import packages should be defined. Any reason for having dynamic (*) imports?

**Pull request merged**